pkgs/applications/science/math/cplex/default.nix

```diff
@@ -0,0 +1,59 @@
+{ stdenv, coreutils, requireFile, patchelf, makeWrapper, openjdk, gtk2, xorg, glibcLocales }:
+
+stdenv.mkDerivation rec {
+  name = "cplex-${version}";
+  version = "128";
+  installer = "cplex_studio${version}.linux-x86-64.bin";
+  
+  src = requireFile rec {
+    name = "${installer}";
+    message = ''
+      This nix expression requires that ${name} is
+      already part of the store. Download it from IBM  
+      and add it to the nix store with nix-store --add-fixed sha256 <FILE>.
+    '';
+    sha256 = "ce8a597a11c73a0a3d49f3fa82930c47b6ac2adf7bc6779ad197ff0355023838";
+  };
+
+  nativeBuildInputs = [ coreutils makeWrapper ];
+  buildInputs = [ openjdk gtk2 xorg.libXtst glibcLocales ];
+
+  phases = "installPhase";
```

Comment:
Don't assign phases directly, instead assign the parts of the script the role they have: The first `sed` is patchPhase, the `sh` buildPhase, put the `ln`'s in installPhase and the `patchelf`/`wrapProgram` in fixupPhase.